backend/inspirehep/tools/views.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020 CERN.
+#
+# inspirehep is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+
+from io import BytesIO, TextIOWrapper
+
+from flask import Blueprint, abort, current_app, jsonify, request
+
+from inspirehep.files.api import current_s3_instance
+from inspirehep.records.utils import hash_data
+from inspirehep.tools.utils import (
+    find_references,
+    get_filename,
+    get_mimetype,
+    get_references,
+)
+
+blueprint = Blueprint("inspirehep_tools", __name__, url_prefix="/tools")
+
+ALLOWED_EXTENSIONS = {"tex"}
+
+
+def allowed_file(filename):
+    return "." in filename and filename.rsplit(".", 1)[1].lower() in ALLOWED_EXTENSIONS
+
+
+@blueprint.route("/bibgen", methods=["POST"])
```

Comment:
why so short? I would suggest something more explicit like
```suggestion
@blueprint.route("/bibliography-generation", methods=["POST"])
```